src/OpenTelemetry.Api/OpenTelemetry.Api.csproj

```diff
@@ -1,11 +1,11 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <!-- OmniSharp/VS Code requires TargetFrameworks to be in descending order for IntelliSense and analysis. -->
-    <TargetFrameworks>netstandard2.0;net462</TargetFrameworks>
+    <TargetFrameworks>net6.0;netstandard2.0;net462</TargetFrameworks>
     <Description>OpenTelemetry .NET API</Description>
     <RootNamespace>OpenTelemetry</RootNamespace>
 
-    <NoWarn>$(NoWarn),CS0618</NoWarn>
+    <NoWarn>$(NoWarn),CS0618, RS0026</NoWarn>
```

Comment:
Why is this necessary?

Reply:
Because of this: https://github.com/open-telemetry/opentelemetry-dotnet/actions/runs/5028125135/jobs/9018449888?pr=4460.
I plan to address these in a separate PR.
```
Error: D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\Baggage.cs(129,59): error RS0026: Symbol 'GetBaggage' violates the backcompat requirement: 'Do not add multiple overloads with optional parameters'. See 'https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md' for details. [D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\OpenTelemetry.Api.csproj::TargetFramework=net6.0]
Error: D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\Baggage.cs(146,30): error RS0026: Symbol 'GetBaggage' violates the backcompat requirement: 'Do not add multiple overloads with optional parameters'. See 'https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md' for details. [D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\OpenTelemetry.Api.csproj::TargetFramework=net6.0]
Error: D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\Baggage.cs(157,31): error RS0026: Symbol 'SetBaggage' violates the backcompat requirement: 'Do not add multiple overloads with optional parameters'. See 'https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md' for details. [D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\OpenTelemetry.Api.csproj::TargetFramework=net6.0]
Error: D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\Baggage.cs(175,31): error RS0026: Symbol 'SetBaggage' violates the backcompat requirement: 'Do not add multiple overloads with optional parameters'. See 'https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md' for details. [D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\OpenTelemetry.Api.csproj::TargetFramework=net6.0]
Error: D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\Trace\Tracer.cs(92,30): error RS0026: Symbol 'StartSpan' violates the backcompat requirement: 'Do not add multiple overloads with optional parameters'. See 'https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md' for details. [D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\OpenTelemetry.Api.csproj::TargetFramework=net6.0]
Error: D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\Trace\Tracer.cs(108,30): error RS0026: Symbol 'StartSpan' violates the backcompat requirement: 'Do not add multiple overloads with optional parameters'. See 'https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md' for details. [D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\OpenTelemetry.Api.csproj::TargetFramework=net6.0]
Error: D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\Trace\Tracer.cs(124,30): error RS0026: Symbol 'StartActiveSpan' violates the backcompat requirement: 'Do not add multiple overloads with optional parameters'. See 'https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md' for details. [D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\OpenTelemetry.Api.csproj::TargetFramework=net6.0]
Error: D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\Trace\Tracer.cs(140,30): error RS0026: Symbol 'StartActiveSpan' violates the backcompat requirement: 'Do not add multiple overloads with optional parameters'. See 'https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md' for details. [D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry.Api\OpenTelemetry.Api.csproj::TargetFramework=net6.0]
```